This notebook measures user satisfaction with their recommendations from the app.

## Import Necessary Libraries

In [1]:
import pymongo
import pandas as pd

## Connect to MongoDB Atlas

The user feedback is stored in MongoDB Atlas. It is retrieved and transformed into a dataframe.

In [2]:
with open('.secrets/password.txt', 'r') as f:
    conn_string = f.read().strip()

In [3]:
mc = pymongo.MongoClient(conn_string)
city_db = mc['city_database']
user_satisfaction = city_db['user_satisfaction']

In [4]:
user_feedback = [x for x in user_satisfaction.find()]
user_feedback[0]

{'_id': ObjectId('5d252e338f3292828b9f6a7a'), 'satisfaction_score': 0.4}

In [6]:
feedback_df = pd.DataFrame(user_feedback)
feedback_df.head()

_id  satisfaction_score
0  5d252e338f3292828b9f6a7a                 0.4
1  5d252e848f3292828b9f6a7c                 0.8
2  5d252f6f8f3292828b9f6a7e                 0.6
3  5d2531bf8f3292828b9f6a82                 1.0
4  5d2532fe8f3292828b9f6a84                 0.6

In [7]:
len(feedback_df)

32

In [9]:
feedback_df.describe()

satisfaction_score
count           32.000000
mean             0.668750
std              0.276426
min              0.000000
25%              0.600000
50%              0.600000
75%              0.850000
max              1.000000

As of 1:48pm on July 10th, 2019 users on average are interested in 67% of their recommended destinations.

Preliminary results show that users are interested in about 67% of their recommendations. However, this number is not reflective of the population as a whole with only 32 samples, which were also subject to sample bias most (if not all) of these samples are from people in my network. Additionally, a few of these samples may be from when I was checking to make sure the website was working.